libraries/botbuilder-dialogs/src/dialogContext.ts

```diff
@@ -2,653 +2,654 @@
-                // Lookup dialog
-                const dialog: Dialog<{}> = this.findDialog(instance.id);
-                if (!dialog) {
-                    throw new DialogContextError(
-                        `DialogContext.repromptDialog(): Can't find a dialog with an id of '${instance.id}'.`,
-                        this
-                    );
-                }
-
-                // Ask dialog to re-prompt if supported
-                await wrapErrors(this, dialog.repromptDialog(this.context, instance));
-            }
-        }
-    }
-
-    /**
-     * Searches for a dialog with a given ID.
-     * @remarks
-     * Emits a named event for the current dialog, or someone who started it, to handle.
-     * @param name Name of the event to raise.
-     * @param value Optional. Value to send along with the event.
-     * @param bubble Optional. Flag to control whether the event should be bubbled to its parent if not handled locally. Defaults to a value of `true`.
-     * @param fromLeaf Optional. Whether the event is emitted from a leaf node.
-     * @returns `true` if the event was handled.
-     */
-    public async emitEvent(name: string, value?: any, bubble = true, fromLeaf = false): Promise<boolean> {
-        // Initialize event
-        const dialogEvent: DialogEvent = {
-            bubble: bubble,
-            name: name,
-            value: value,
-        };
-
-        // Find starting dialog
-        let dc: DialogContext = this;
-        if (fromLeaf) {
-            while (true) {
-                const childDc = dc.child;
-                if (childDc != undefined) {
-                    dc = childDc;
-                } else {
-                    break;
-                }
-            }
-        }
-
-        // Dispatch to active dialog first
-        // - The active dialog will decide if it should bubble the event to its parent.
-        const instance = dc.activeDialog;
-        if (instance != undefined) {
-            const dialog = dc.findDialog(instance.id);
-            if (dialog != undefined) {
-                return wrapErrors(this, dialog.onDialogEvent(dc, dialogEvent));
-            }
-        }
-
-        return false;
-    }
-
-    /**
-     * @private
-     * @param reason
-     * @param result
-     */
-    private async endActiveDialog(reason: DialogReason, result?: any): Promise<void> {
-        const instance: DialogInstance<any> = this.activeDialog;
-        if (instance) {
-            // Lookup dialog
-            const dialog: Dialog<{}> = this.findDialog(instance.id);
-            if (dialog) {
-                // Notify dialog of end
-                await wrapErrors(this, dialog.endDialog(this.context, instance, reason));
-            }
-
-            // Pop dialog off stack
-            this.stack.pop();
-
-            this.state.setValue(TurnPath.lastResult, result);
-        }
-    }
-}
+ import { Activity, TurnContext, TurnContextStateCollection } from 'botbuilder-core';
```

Comment:
This entire file is not properly fixed, why is this file changed from another library?